### DISCUSSION TITLE
Add a prometheus metric gauge for Datasource/Panel references

### DISCUSSION THREAD

**Discussion by appleciderjack** [grafana/grafana#38143] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
A counter in the prometheus metric route that specifies how many references there are for each data sources.
Example:
grafana_datasource_reference_count{name="prometheus 1", type="prometheus"} 10
grafana_datasource_reference_count{name="old Graphite", type="graphite"} 0

**Why is this needed**:
We have old data sources that are referenced in old panels that we would like to clean out. It would be great to know when removing data sources that it is not being referred to by any current version of the panels.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
